src/app/code/community/Zendesk/Zendesk/Model/Observer.php

```diff
@@ -97,6 +97,20 @@ public function saveConfig(Varien_Event_Observer $observer)
                 Mage::getModel('core/config')->saveConfig('contacts/email/recipient_email', $oldEmail, $scope, $scopeId);
             }
         }
+
+        // If the zendesk domain is not found in the web widget snippet (wrapped with quotes), generate it again
+        $zDomain = Mage::getStoreConfig('zendesk/general/domain', $storeCode);
+        $widgetSnippet = Mage::getStoreConfig('zendesk/frontend_features/web_widget_code_snippet', $storeCode);
+        // Case insensitive search with single and double quotes, still better performance than 1 regexp search
+        if(stripos($widgetSnippet, "'{$zDomain}'") === false && stripos($widgetSnippet, '"'.$zDomain.'"') === false) {
+            $webWidgetSnippet=<<<EOJS
+<!-- Start of Zendesk Widget script -->
+<script>/*<![CDATA[*/window.zEmbed||function(e,t){var n,o,d,i,s,a=[],r=document.createElement("iframe");window.zEmbed=function(){a.push(arguments)},window.zE=window.zE||window.zEmbed,r.src="javascript:false",r.title="",r.role="presentation",(r.frameElement||r).style.cssText="display: none",d=document.getElementsByTagName("script"),d=d[d.length-1],d.parentNode.insertBefore(r,d),i=r.contentWindow,s=i.document;try{o=s}catch(c){n=document.domain,r.src='javascript:var d=document.open();d.domain="'+n+'";void(0);',o=s}o.open()._l=function(){var o=this.createElement("script");n&&(this.domain=n),o.id="js-iframe-async",o.src=e,this.t=+new Date,this.zendeskHost=t,this.zEQueue=a,this.body.appendChild(o)},o.write('<body onload="document._l();">'),o.close()}("//assets.zendesk.com/embeddable_framework/main.js","{$zDomain}");/*]]>*/</script>
+<!-- End of Zendesk Widget script -->
+EOJS;
+
+            Mage::getModel('core/config')->saveConfig('zendesk/frontend_features/web_widget_code_snippet', $webWidgetSnippet);
```

Comment:
Hey Mio, my understanding is that the `core/config` model contains all the logic to manage/CRUD the config stored in the DB: read config values, set/update them, and delete them. Our config is all under they key `zendesk/*`.
We could create another model to read the config... but the one already provided by Magento does everything we need, and provides events that you can react to, like here: "do something if the config is updated".